Cython/Compiler/ExprNodes.py

```diff
@@ -2204,6 +2204,13 @@ def analyse_types(self, env):
         entry = self.entry
         if entry is None:
             entry = env.lookup(self.name)
+            if env.analysing_evaluated_annotation:
+                if self.analyse_as_type(env) and (
+                        not entry or (entry.is_type and not entry.as_variable)):
+                    # it's a C type in an annotation. It can't be sensibly evaluated
+                    # so return a string instead. The annotation node will be prepared
+                    # to handle this exception
+                    raise AnnotationNode.NeedsStringConversion
```

Comment:
This isn't the best thing ever, but it seemed the easiest way of abandoning the analysis and jumping upwards